lib/src/clojure_lsp/feature/move_form.clj

```diff
@@ -0,0 +1,203 @@
+                                            (= (:name %) source-ns))
+                                      local-analysis))
+        remove-source-require? (and source-require (empty? other-source-usages))
+        namespace-loc (edit/find-namespace file-loc)]
+    (if-let [add-to-ns-changes (f.add-missing-libspec/add-to-namespace* file-loc libspec db)]
+      (cond-> add-to-ns-changes
+        remove-source-require?
+        (update-in
+          [0 :loc]
+          (fn [loc]
+            (z/subedit->
+              loc
+              (edit/find-at-usage source-require)
+              z/up
+              z/remove)))
+
+        (and (not remove-source-require?) source-refer)
+        (update-in
+          [0 :loc]
+          (fn [loc]
+            (z/subedit->
+              loc
+              (edit/find-at-usage-name source-refer)
+              z/remove
+              (cond-> (empty? other-source-refers) (-> z/remove z/remove)))))
+
+        :always
+        (->> (f.add-missing-libspec/cleaning-ns-edits uri db)))
+      (when (or remove-source-require? source-refer)
+        (->> [{:loc (cond-> namespace-loc
+                      remove-source-require?
+                      (z/subedit->
+                        (edit/find-at-usage source-require)
+                        z/up
+                        z/remove)
+
+                      (and (not remove-source-require?) source-refer)
+                      (z/subedit->
+                        (edit/find-at-usage-name source-refer)
+                        z/remove
+                        (cond-> (empty? other-source-refers) (-> z/remove z/remove))))
+               :range (meta (z/node namespace-loc))}]
+             (f.add-missing-libspec/cleaning-ns-edits uri db))))))
+
+(defn move-form [zloc uri db dest-filename]
+  (let [form-loc (edit/to-top zloc)
+        analysis (:analysis @db)
+        source-filename (shared/uri->filename uri)
+        source-ns (:name (q/find-namespace-definition-by-filename analysis source-filename db))
+        dest-filename (shared/absolute-path dest-filename db)
+        dest-ns (:name (q/find-namespace-definition-by-filename analysis dest-filename db))
+        inner-usages (var-usages-within zloc uri db)
+        ;; if source-ns things are used within the form
+        ;; we can't move it
+        local-inner-usages (->> inner-usages
+                                (filterv (comp #(= source-ns %) :to)))
+        defs (var-definitions-within zloc uri db)
+        on-top-level-form? (= form-loc zloc)
+        ;; You could have multiple in a top form (let [x 1] (def y x) (def z y))
+        ;; we don't support that
+        multiple-defs? (= 1 (count defs))
+        can-move? (and dest-ns
+                       (empty? local-inner-usages)
+                       on-top-level-form?
+                       multiple-defs?)]
+    (when can-move?
+      (let [def-to-move (first defs)
+            refs (q/find-references analysis def-to-move false true)
```

Comment:
What is this `true`? shouldn't we pass db there?

Reply:
I think I had started down the path of only looking for project refs. I see @mainej has added something similar, and this can be updated later if desired. It's generally less important as it's ok imo, for this command to take slightly longer.

Reply:
@snoe I mean, that find-references function, receives as last arg a `db` right? Why we are passing true instead of a db component?

Reply:
right, it was a good catch, db is being passed now.